sqs/src/main/scala/akka/stream/alpakka/sqs/scaladsl/SqsAckFlow.scala

```diff
@@ -20,45 +26,169 @@ import scala.concurrent.Future
+      implicit sqsClient: SqsAsyncClient
+  ): Flow[MessageAction, SqsAckResult[SqsResponse], NotUsed] =
+    Flow[MessageAction]
+      .mapAsync(settings.maxInFlight) {
+        case messageAction: MessageAction.Delete =>
+          val request =
+            DeleteMessageRequest
+              .builder()
+              .queueUrl(queueUrl)
+              .receiptHandle(messageAction.message.receiptHandle())
+              .build()
+
+          sqsClient
+            .deleteMessage(request)
+            .toScala
+            .map(resp => new SqsAckResult(Some(resp: SqsResponse), messageAction))(sameThreadExecutionContext)
+
+        case messageAction: MessageAction.ChangeMessageVisibility =>
+          val request =
+            ChangeMessageVisibilityRequest
+              .builder()
+              .queueUrl(queueUrl)
+              .receiptHandle(messageAction.message.receiptHandle())
+              .visibilityTimeout(messageAction.visibilityTimeout)
+              .build()
+
+          sqsClient
+            .changeMessageVisibility(request)
+            .toScala
+            .map(resp => new SqsAckResult(Some(resp: SqsResponse), messageAction))(sameThreadExecutionContext)
+
+        case messageAction: MessageAction.Ignore =>
+          Future.successful(new SqsAckResult[SqsResponse](None, messageAction))
+      }
 
+  /**
+   * creates a [[akka.stream.scaladsl.Flow Flow]] for ack grouped SQS messages using an [[software.amazon.awssdk.services.sqs.SqsAsyncClient]].
+   */
   def grouped(queueUrl: String, settings: SqsAckGroupedSettings = SqsAckGroupedSettings.Defaults)(
-      implicit sqsClient: AmazonSQSAsync
-  ): Flow[MessageAction, SqsAckResult, NotUsed] = {
-    val graph = GraphDSL.create() { implicit builder =>
-      import GraphDSL.Implicits._
+      implicit sqsClient: SqsAsyncClient
+  ): Flow[MessageAction, SqsAckResult[SqsResponse], NotUsed] =
+    Flow.fromGraph(
+      GraphDSL.create() { implicit builder =>
+        import GraphDSL.Implicits._
+
+        val p = builder.add(Partition[MessageAction](3, {
+          case _: Delete => 0
+          case _: ChangeMessageVisibility => 1
+          case _: MessageAction.Ignore => 2
+        }))
+
+        val merge = builder.add(Merge[SqsAckResult[SqsResponse]](3))
+
+        val mapDelete = Flow[MessageAction].collectType[Delete]
+        val mapChangeMessageVisibility = Flow[MessageAction].collectType[ChangeMessageVisibility]
 
-      val partitionByAction = builder.add(Partition[MessageAction](3, {
-        case _: MessageAction.Delete => 0
-        case _: MessageAction.ChangeMessageVisibility => 1
-        case _: MessageAction.Ignore => 2
-      }))
+        p.out(0) ~> mapDelete ~> groupedDelete(queueUrl, settings) ~> merge
+        p.out(1) ~> mapChangeMessageVisibility ~> groupedChangeMessageVisibility(queueUrl, settings) ~> merge
+        p.out(2) ~> Flow[MessageAction].map(x => new SqsAckResult[SqsResponse](None, x)) ~> merge
```

Comment:
Ahh, this is a nice use-case for yet-to-be-merged https://github.com/akka/akka/pull/26280

Reply:
fixed